ci failures for BearerTokenAuthorizationTest

**Describe the bug**
CI native build for the mentioned test fails
**Expected behavior**
CI to pass

**Screenshots**
```
INFO] Running io.quarkus.it.keycloak.BearerTokenAuthorizationTest
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.282 s <<< FAILURE! - in io.quarkus.it.keycloak.BearerTokenAuthorizationTest
[ERROR] io.quarkus.it.keycloak.BearerTokenAuthorizationTest  Time elapsed: 4.282 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <201> but was <500>.

	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
```

**Additional context**
the following PRs from https://github.com/quarkusio/quarkus/pull/4609 to https://github.com/quarkusio/quarkus/pull/4623 seems to have hit this failure.

--- a/ci-templates/native-build-steps.yaml
+++ b/ci-templates/native-build-steps.yaml
@@ -29,7 +29,7 @@ jobs:
       - script: docker run --rm --publish 8000:8000 --name build-dynamodb -d amazon/dynamodb-local:1.11.477
         displayName: 'start dynamodb'
     - ${{ if eq(parameters.keycloak, 'true') }}:
-        - script: docker run --rm --publish 8180:8080 --name build-keycloak  -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin -d quay.io/keycloak/keycloak
+        - script: docker run --rm --publish 8180:8080 --name build-keycloak  -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin -d quay.io/keycloak/keycloak:7.0.0
           displayName: 'start keycloak'
     - ${{ if eq(parameters.mysql, 'true') }}:
         - bash: |

